PixaPicker/PixaFavoritesViewController.swift

```diff
@@ -0,0 +1,105 @@
+//
+//  PixaFavoritesViewController.swift
+//  PixaPicker
+//
+//  Created by Thomas Caraway on 2/3/20.
+//  Copyright © 2020 Thomas Caraway. All rights reserved.
+//
+
+import Foundation
+import SDWebImage
+import UIKit
+import CoreData
+
+
+//NOTE: WHEN REMOVING PICS FROM FAVES, WILL ALSO REMOVE ANY DUPLICATES. WONT BE ISSUE ONCE WE DISABLE FAVORITE BUTTON AFTER IT GETS PRESSED IN SEARCHVIEWCONTROLLER
+class PixaFavoritesViewController: UIViewController, PixaSaveButtonDelegate, PixaFavoriteButtonDelegate, UICollectionViewDataSource, UICollectionViewDelegate, UICollectionViewDelegateFlowLayout, PixaFavoritesCoordinatorDelegate{
+    
+    @IBOutlet weak var imageCollectionView: UICollectionView!
+    let reuseID = "cell"
+    let favoritesCoordinator = PixaFavoritesCoordinator()
+    var delegate: PixaViewControllerDelegate?
+    let testString = "FAVES CONTROLLER EXISTS"
+    
+    //UICollectionViewDataSource protocol functions
+    func collectionView(_ collectionView: UICollectionView, numberOfItemsInSection section: Int) -> Int {
+        return favoritesCoordinator.cellImageURLs.count
+    }
+    
+    func collectionView(_ collectionView: UICollectionView, cellForItemAt indexPath: IndexPath) -> UICollectionViewCell {
+        
+        let imageCell = collectionView.dequeueReusableCell(withReuseIdentifier: reuseID, for: indexPath as IndexPath) as! PixaCollectionViewCell
+        
+        imageCell.delegate = self
+        imageCell.delegate2 = self
+        
+        let imageURL = URL(string: favoritesCoordinator.cellImageURLs[indexPath.row])
+        imageCell.cellImage.sd_setImage(with: imageURL, placeholderImage: nil)
```

Comment:
same here.  the `dataSource` protocol could have a `dataSource.getUrl(for index: indexPath.row) -> URL?` which the `FavoritesDataCoordinator` can inherit and provide the data for.

Reply:
Imagine if you wanted to use this `ViewController` again but in a different context.  You would have to rip out the `favoritesCoordinator` and plug in something new.  a `protocol FavoritesViewControllerDataSource ...` will help you make less assumptions about how it will be used